src/main/java/org/jboss/pnc/bifrost/source/db/LogEntry.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * JBoss, Home of Professional Open Source.
+ * Copyright 2020-2022 Red Hat, Inc., and individual contributors
+ * as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jboss.pnc.bifrost.source.db;
+
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import io.quarkus.hibernate.orm.panache.PanacheEntityBase;
+import io.quarkus.panache.common.Parameters;
+import lombok.AllArgsConstructor;
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.Setter;
+import lombok.ToString;
+
+import javax.persistence.Cacheable;
+import javax.persistence.Column;
+import javax.persistence.Entity;
+import javax.persistence.Id;
+import javax.persistence.Index;
+import javax.persistence.NamedQuery;
+import javax.persistence.QueryHint;
+import javax.persistence.Table;
+import javax.validation.constraints.NotNull;
+import java.util.Optional;
+
+@EqualsAndHashCode(exclude = "id")
+@JsonIgnoreProperties(ignoreUnknown = true)
+@Getter
+@Setter
+@AllArgsConstructor
+@NoArgsConstructor
+@Entity
+@ToString
+@Table(
+        indexes = {
+                @Index(name = "idx_logentry_processContext", columnList = "processContext"),
+                @Index(name = "idx_logentry_processContextVariant", columnList = "processContextVariant"),
+                @Index(name = "idx_logentry_requestContext", columnList = "requestContext"),
+                @Index(name = "idx_logentry_buildId", columnList = "buildId") })
+@NamedQuery(
+        name = "LogEntry.findExisting",
+        query = "FROM LogEntry e" + " WHERE e.processContext = :processContext"
+                + "   AND e.processContextVariant = :processContextVariant"
+                + "   AND e.requestContext = :requestContext" + "   AND e.temporary = :temporary"
+                + "   AND e.buildId = :buildId",
+        hints = @QueryHint(name = "org.hibernate.cacheable", value = "true"))
+@Cacheable
+public class LogEntry extends PanacheEntityBase {
+
+    @Id
```

Comment:
Can you please use a sequence generator:
    @Id
    @SequenceGenerator(
            name = "logEntrySequence",
            sequenceName = "logentry_id_sequence",
            allocationSize = 1,
            initialValue = 1000000)
    @GeneratedValue(strategy = GenerationType.SEQUENCE, generator = "logEntrySequence")

P.S. initial value is an estimation, could be refined later

Reply:
It's intentional that the ID is generated in the application as I believe generating the "globally" unique ID (same as in other PNC services) in the application performs better:
- no need for another IO operation to get the ID from the DB
- no need to sync the sequence in the DB cluster (geo disrtibution)

Reply:
ah ok, makes sense, I was afraid it was going to be an autoincrement id. ok so the logentry.id should be mapped as bigint also.